pkg/cluster/spec/prometheus.go

```diff
@@ -28,23 +28,31 @@ import (
 	"github.com/pingcap/tiup/pkg/cluster/template/scripts"
 	"github.com/pingcap/tiup/pkg/meta"
 	"github.com/pingcap/tiup/pkg/set"
+	"gopkg.in/yaml.v2"
 )
 
 // PrometheusSpec represents the Prometheus Server topology specification in topology.yaml
 type PrometheusSpec struct {
-	Host            string               `yaml:"host"`
-	SSHPort         int                  `yaml:"ssh_port,omitempty" validate:"ssh_port:editable"`
-	Imported        bool                 `yaml:"imported,omitempty"`
-	Port            int                  `yaml:"port" default:"9090"`
-	DeployDir       string               `yaml:"deploy_dir,omitempty"`
-	DataDir         string               `yaml:"data_dir,omitempty"`
-	LogDir          string               `yaml:"log_dir,omitempty"`
-	NumaNode        string               `yaml:"numa_node,omitempty" validate:"numa_node:editable"`
-	Retention       string               `yaml:"storage_retention,omitempty" validate:"storage_retention:editable"`
-	ResourceControl meta.ResourceControl `yaml:"resource_control,omitempty" validate:"resource_control:editable"`
-	Arch            string               `yaml:"arch,omitempty"`
-	OS              string               `yaml:"os,omitempty"`
-	RuleDir         string               `yaml:"rule_dir,omitempty" validate:"rule_dir:editable"`
+	Host            string                 `yaml:"host"`
+	SSHPort         int                    `yaml:"ssh_port,omitempty" validate:"ssh_port:editable"`
+	Imported        bool                   `yaml:"imported,omitempty"`
+	Port            int                    `yaml:"port" default:"9090"`
+	DeployDir       string                 `yaml:"deploy_dir,omitempty"`
+	DataDir         string                 `yaml:"data_dir,omitempty"`
+	LogDir          string                 `yaml:"log_dir,omitempty"`
+	NumaNode        string                 `yaml:"numa_node,omitempty" validate:"numa_node:editable"`
+	RemoteConfig    map[string]interface{} `yaml:"remote_config,omitempty" validate:"remote_config:ignore"`
```

Comment:
Why not use

```go
RemoteConfig    Remote `yaml:"remote_config,omitempty" validate:"remote_config:ignore"`
```

By this way, you don't need to marshal and unmarshal, what the user input is ensured a valid yaml...

Reply:
It depends on the user，if the user config like below
```
monitoring_servers:
- host: 10.4.45.162
  ssh_port: 22
  imported: true
  port: 9090
  deploy_dir: /data/deploy
  data_dir: /data/deploy/prometheus2.0.0.data.metrics
  log_dir: /data/deploy/logremote_config:
  remote_config:
    remote_write:
    - queue_config:
        batch_send_deadline: 5m
        capacity: 100000
        max_samples_per_send: 10000
        max_shards: 300
      url: http://127.0.0.1://8003/write
    remote_read:
    - url: http://127.0.0.1://8003/read
    global:
      ****
    rule_files:
    - ***
```
So, I add marshal and unmarshal to remove the redundant configuration.

Reply:
You are right and I had fixed it.